File: 525-ContiguousArray/525-ContiguousArray.cpp
// Last updated: 02/08/2026, 14:32:08
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int, int> mp;
        mp[0] = -1;
        int prefix = 0;
        int result = 0;

        for (int i = 0; i < nums.size(); ++i){
            prefix += (nums[i] == 0) ? -1 : 1;

            if (mp.find(prefix) != mp.end()){
                result = max(result, i - mp[prefix]);
            }
            else mp[prefix] = i;
        }
        return result;
    }
};

File: 525-ContiguousArray/525-ContiguousArray.py
# Last updated: 02/08/2026, 14:32:04
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        mapping = {}
        mapping[0] = -1
        prefix, result = 0, 0
        for i, val in enumerate(nums):
            prefix += -1 if val == 0 else 1
            if (prefix in mapping):
                result = max(result, i - mapping[prefix])
            else:
                mapping[prefix] = i
        return result